clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -163,6 +163,15 @@ else if (maxTimestamp == LogEntry.NO_TIMESTAMP)
             return new RecordsInfo(maxTimestamp, compressionType == CompressionType.NONE ? offsetOfMaxTimestamp : lastOffset);
     }
 
+    /**
+     * We set the sequence of the batch just before it is about to be sent out.
+     * @param baseSequence
+     */
+    public void closeWithSequence(int baseSequence) {
+        this.baseSequence = baseSequence;
```

Comment:
I guess we might want to be a little careful that the record set hasn't already been closed. Perhaps we can raise an `IllegalStateException` if `builtRecords` is not null?